clients/src/test/java/org/apache/kafka/common/requests/FetchRequestTest.java

```diff
@@ -243,4 +244,43 @@ public void testPartitionDataEquals() {
             new FetchRequest.PartitionData(Uuid.randomUuid(), 300, 0L, 300, Optional.of(300)));
     }
 
+    @Test
+    public void testFetchRequestNoCacheData() {
+        short version = 13;
```

Comment:
Do we have test case for version < 13?

Reply:
Yes, in `testToReplaceWithDifferentVersions` test all version

Reply:
My point was that we should have test to check the `fetchRequest.forgottenTopics` for all versions. `testToReplaceWithDifferentVersions` does NOT do that

Reply:
I test all version for `testFetchRequestNoCacheData`, PTAL